### ISSUE TITLE
MUFI 3.0/4.0 glyphs not copying/displaying correctly

### ISSUE THREAD

**Issue by kreddy95** [WheatonCS/Lexos#397]
### which file did I use
in the resources folder -> MUFI_3_DICT.tsv or MUFI_4_DICT.tsv

### what the bug looks like
Some of the MUFI characters (glyphs) arn't being copied as the correct characters from the official MUFI pdf site into the .tsv files for both MUFI_3_DICT.tsv and MUFI_4_DICT.tsv
I found one example in the MUFI_3_FULL.xlsx file where the image isnt being copied correctly (line 148 &ains; (also in MUFI_4_FULL.xlsx but on a different line) ) into that file but all the other ones show up as the boxes so it seems like nothing is wrong until the .tsv file is opened 

### what I expect to happen
the correct glyphs should be displayed when the file is opened matching the official tables

### how to reproduce it
go to the resources file and open the MUFI_3_DICT.tsv or the MUFI_4_DICT.tsv files and look at the first column 
Here are the screenshots of what it is displaying and one of what the official document has as the glyphs so this is easier to picture what is happening:
![mufi_3_dict_notdisplaycorrectly](https://cloud.githubusercontent.com/assets/19455351/16389316/4e717e6e-3c6b-11e6-96c6-92730d0673e8.png)
![mufi_3_official](https://cloud.githubusercontent.com/assets/19455351/16389518/4e908e34-3c6c-11e6-81a9-2570156c6c48.png)

for example: in MUFI_3_DICT.tsv
line 140: &aesup; shows up as an air plane where as the official document shows it as a lowercase a with an e above it 
line 145: &avsup; shows up as a snowflake where as the official documents shows it as a lowercase a with a v above it
line 148: &ains; shows up as ubuntu with a superscript circle next to the last u where as the official document shows it as a type of a 

(There are alot more of these throughout the entire file that arn't showing up as they should these are only a few examples, more can be seen just by looking at the screenshot as well)

**Comment by scottkleinman** [comment]
I'll think about this one as soon as I can, but you might get to it first, as I'm busy tackling Cutter. But this certainly would explain the irregular conversions that you've been noticing. Is `&aeligdotacute;` one of the characters that is copying wrong? That was one that I remember failing to convert...

**Comment by kreddy95** [comment]
yes &aeligdotacute; shows up as a m but in the official one it shows it as a an the AE with a dot and acute above it and &AEligdotacute; shows up as a N but on the official site its an ae with a dot and acute above it but these two are only in MUFI 4.0
(im not sure if the official site mixed those two up based on the the names it seems like they should be flipped)

**Comment by scottkleinman** [comment]
I think the closest we are going to get to a solution is to go through a list like [this one](http://www.fileformat.info/info/charset/UTF-8/list.htm) and copy and paste from there. I think, if you generate a list of the characters that fail, we can do this gradually after the chaos of this week ends.

**Comment by kreddy95** [comment]
Im not sure how we'd generate a list from that site, some of the characters aren't even on there I tried searching for a few that arent working and the site has no results for them. I also just tried opening the .tsv file on a windows computer and everything showed up even worse on there than on the linux box that im using.

**Comment by kreddy95** [comment]
once we installed junicode that made it so that the glyphs displayed correctly for the few I was looking at in 3.0 (just from scrolling through it quick we believe for most of them if not all of the file as well) but only when that font is used

**Comment by mleblanc321** [comment]
nice to see;  so, we'll update the ITM.md file accordingly, reminding us that Middle English users who want MUFI3/4 _must_ have Junicode (or some applicable font) installed;  (but, this makes sense to me; wouldn't these users of Middle English already have such a font installed?)

**Comment by scottkleinman** [comment]
Well, that's encouraging. It's simply the case that the standard allows you to encode characters regardless of whether fonts exist to display them. Users can be expected to have whatever fonts are necessary to display there characters. But see my suggestion in issue #365. I have successfully displayed the results in Junicode in Lexos, but I took that code out since we don't want it to be a default font. We could perhaps set the preview css to Junicode automatically if the user selects MUFI conversion.

**Comment by chantisnake** [comment]
I think we can give them a zip pack of the judecode, ttf font is super easy to install on windows. I will test if that works

Also we need to change the frontend to use judecode as the default font on the preview.

**Comment by mleblanc321** [comment]
i'm not convinced we should add junicode to repo;  perhaps we should consider this addition (by us rather than the user) to be the user's responsibility for this v3.0 release?

**Comment by chantisnake** [comment]
We can also put a little message on the scrub page saying if your text is not displaying correctly, click this.

And then link to the instruction to install and use judocode on the browser

We should also tell them, all the analysis result will be correct regardless of what is displayed.

**Comment by chantisnake** [comment]
Yes, just add a zip pack in the release or some where.

**Comment by mleblanc321** [comment]
>We can also put a little message on the scrub page saying if your text is not displaying correctly, click this.And then link to the instruction to install and use judocode on the browser
We should also tell them, all the analysis result will be correct regardless of what is displayed.

yes, yes, yes; very good;  (i thought you went to dinner? :)

**Comment by scottkleinman** [comment]
I'm not sure that is a good idea. Junicode is specifically for medieval European characters, so it's extra baggage for anyone not working with medieval European texts. So I think it's better to provide a download link in ITM for anyone who doesn't have it on their system. Additionally, it's possible to load the font into the css over the internet, so users don't absolutely have to have Junicode as long as they have an internet connection. So I think a better way would be to automate it would be to pass a variable saying that the users have converted MUFI entities. Let that trigger `$(".filecontents").addClass("Junicode")`. That worked successfully when I tested it a couple of days ago.

**Comment by mleblanc321** [comment]
right;  i'm hoping kristina will stay on this ...

**Comment by chantisnake** [comment]
I think we only show that line when people select mufi.

**Comment by chantisnake** [comment]
Or we can say what exactly does this font do in the instructions (medial Europe text)

**Comment by scottkleinman** [comment]
I've implemented a version of what we need. If you select one of the MUFI options, the Previews window will automatically switch to Junicode. However, this does require having Junicode embedded in the web page. I've appended a `<link>` element to `<head>` when the user selects a MUFI option. (I didn't really think this would work, but it did.) The font is in the `static/lib` folder created for Local Mode. This avoids loading the font in `style.css` when most users won't need it. It does mean that the font will stay loaded on the page if the user opts for something other than MUFI, but I don't think that's really a problem. The main downside, is that we must make Junicode part of the Lexos installation.

Eventually, we may want to give the user the ability to enable and disable fonts through the settings menu, but that's much further down the line.
